frontend/src/pages/modelServing/utils.ts

```diff
@@ -211,3 +212,20 @@ export const isModelServerEditInfoChanged = (
           createData.tokens.map((token) => token.name).sort(),
         ))
     : true;
+
+export const checkPlatformAvailability = (
+  status: DataScienceClusterKindStatus,
+): { kServeAvailable: boolean; modelMeshAvailable: boolean } => ({
+  kServeAvailable: status.installedComponents.kserve,
+  modelMeshAvailable: status.installedComponents['model-mesh'],
```

Comment:
When the status is failed, these 2 values could be `undefined` in `installedComponents`, so we should try to make it a boolean.
```suggestion
  kServeAvailable: !!status.installedComponents.kserve,
  modelMeshAvailable: !!status.installedComponents['model-mesh'],
```

Reply:
Can we also update the type to have optionals?

```
installedComponents: { [key in ComponentNames]?: boolean };
```

Let us be intentful in all aspects of TypeScript. Don't assume any states.

Reply:
Yes, that's right, I bumped into that same question @andrewballantyne when creating the mocks for DSC, but I assume they were mandatory, I agree that we shouldn't do that.